src/MessageRouter.py

```diff
@@ -0,0 +1,47 @@
+import time
+import re
+
+
+class MessageRouter:
+    """
+    Responsible for invoking a message command handler if a message contains
+    a command and the rate limit is not exceeded.
+    """
+
+    def __init__(self,
+                 command_handlers={},
+                 default_rate_limit_in_seconds=60,
+                 per_command_rate_limit_in_seconds={}):
+        self._default_rate_limit = default_rate_limit_in_seconds
+        self._command_handlers = command_handlers
+        self._per_command_rate_limit = per_command_rate_limit_in_seconds
+
+        self._time_of_last_command_usage = {}
+
+    def handle_message(self, message):
```

Comment:
Did you misplace this comment? MessageRouter has no reference to `#aeolus` and indeed does not even know that it is, in case of the production code, used for IRC stuff.

The only occurrence of `#aeolus` in code I touched is in `BotModeration.on_pubmsg`. I changed the code in such a way that it no longer binds to the IRC library and this string in dozens of places. I did not make the channel configurable, which would have been an additional change. Perhaps that is a good change and perhaps it should be made. That's out of scope of what I am trying to do here though. And my changes make it far easier to make the channel configurable in a sane way.

Or did I miss something?